plugins/coverimage.koplugin/main.lua

```diff
@@ -0,0 +1,190 @@
+local Device = require("device")
+
+if not Device.isAndroid() and not Device.isEmulator() then
+    return { disabled = true }
+end
+
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local CoverImage = WidgetContainer:new{
+    name = 'coverimage',
+    is_doc_only = true,
+    settings_id = 0,
+}
+
+function CoverImage:_enabled()
+    return self.enabled
+end
+
+function CoverImage:_restore()
+    return self.enabled and self.restore
+end
+
+function CoverImage:init()
+    self.cover_image_path = G_reader_settings:readSetting("cover_image_path") or "Cover.png"
+    self.enabled = G_reader_settings:isTrue("cover_image_enabled")
+    self.restore = G_reader_settings:isTrue("cover_image_restore") and self.enabled
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function CoverImage:cleanUpImage()
+    if not self.ui.doc_settings:readSetting("exclude_cover_image") == true then
+        local lfs = require("libs/libkoreader-lfs")
+        -- Delete image the backup (if it exits) will contain a user defined sleep image
+        if lfs.attributes(self.cover_image_path) then
+            os.remove(self.cover_image_path)
+        end
+        -- On Tolino a user defined /sdcard/suspend_others.jgp can be used as screensaver on system sleep.
+        -- Restore backup if it exists, so we can use it as sleep image
+        -- If no backup exists, there is no file and the native sleep image is used
+        local bak_file_name = self.cover_image_path .. ".bak"
+        if lfs.attributes(bak_file_name) then
+            os.rename(bak_file_name, self.cover_image_path)
+        end
+    end
+end
+
+function CoverImage:onCloseDocument()
+    logger.dbg("CoverImage: onCloseDocument")
+    -- Try to restore the state before opening of a document.
+    if self.restore then
+        self:cleanUpImage()
+    end
+end
+
+function CoverImage:onReaderReady(doc_settings)
+   logger.dbg("CoverImage: onReaderReady")
+   if self.enabled and not doc_settings:readSetting("exclude_cover_image") == true then
+        local lfs = require("libs/libkoreader-lfs")
+        local image = self.ui.document:getCoverPageImage()
+        if image then
```

Comment:
Still an issue in onCloseDocument/cleanUpImage() if there were no image here: the one that is `os.remove(self.cover_image_path)` is actually the main fallback image.